arithmetic/src/addition/csa/common/CSACompressor5_3.scala

```diff
@@ -0,0 +1,86 @@
+package arithmetic.addition.csa.common
+
+import arithmetic.addition.csa._
+import chisel3._
+
+object CSACompressor5_3 extends CSACompressor(5, 3) {
+  val encodeTable: Map[BigInt, Seq[BigInt]] = tableBuilder(
+    Map(
+      // 0
+      "000" -> Seq(
+        "00000"
+      ),
+      // 1
+      "001" -> Seq(
+        "10000",
+        "01000",
+        "00100",
+        "00010",
+        "00001"
+      ),
+      // 2
+      "010" -> Seq(
+        // count(012) = 2
+        "11000",
+        "01100",
+        "10100"
+      ),
+      "100" -> Seq(
+        // count(012) = 1 count(34) = 1
+        "10010",
+        "01010",
+        "00110",
+        "10001",
+        "01001",
+        "00101",
+        // count(012) = 0
+        "00011"
```

Comment:
I have been thinking, is possible to make this encode be more balance like 5-5, while currently 7-3.